ccvm/ccvm.go

```diff
@@ -170,41 +170,57 @@ func StartFlags() (VMSpec, error) {
 	return customSpec, nil
 }
 
-// Create sets up the VM
-func Create(ctx context.Context, workloadName string, debug bool, update bool, customSpec *VMSpec) error {
-	var err error
-
+func prepareCreate(ctx context.Context, workloadName string, debug bool, update bool, customSpec *VMSpec) (*workload, *workspace, error) {
 	ws, err := prepareEnv(ctx)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	wkld, err := createWorkload(ctx, ws, workloadName)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	in := &wkld.spec.VM
 
 	err = in.mergeCustom(customSpec)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	ws.Mounts = in.Mounts
 	ws.Hostname = wkld.spec.Hostname
 	if ws.NoProxy != "" {
 		ws.NoProxy = fmt.Sprintf("%s,%s", ws.Hostname, ws.NoProxy)
-	} else if ws.HTTPProxy != "" || ws.HTTPSProxy != "" {
-		ws.NoProxy = ws.Hostname
+		ws.PackageUpgrade = "false"
```

Comment:
It looks like there's been some sort of merge issue here.  This block of code is duplicated after the if statement.

Also, there's a change in behaviour here.  Previously, if no_proxy was not defined but http_proxy or https_proxy were, ccloudvm was initialising ws.NoProxy to ws.Hostname.  The code no longer does this.